sdk/inc/azure/iot/az_iot_adu_client.h

```diff
@@ -29,6 +29,11 @@
 
 #include <azure/core/_az_cfg_prefix.h>
 
+/**
+ * @brief  Define the ADU agent interface ID.
+ */
+#define AZ_IOT_ADU_CLIENT_AGENT_INTERFACE_ID "dtmi:azure:iot:deviceUpdate;1"
```

Comment:
Instead of exposing the interface ID, should we rather offer a more flexible API that would alter the `az_iot_hub_client_options`? 

That would allow us in the future to change other things and not tie in the application to this particular constant.
The new function would not alter an already existing PnP model and would only set it if the application didn't configure it before.